pkg/tfvars/libvirt/cache.go

```diff
@@ -44,13 +44,14 @@ func (libvirt *Libvirt) UseCachedImage() (err error) {
 		return err
 	}
 
+	// Get the etag with HEAD
 	cache := diskcache.New(httpCacheDir)
 	transport := httpcache.NewTransport(cache)
-	resp, err := transport.Client().Get(libvirt.Image)
+	resp, err := transport.Client().Head(libvirt.Image)
```

Comment:
My understanding of the cache library was that when the upstream image hadn't changed, it would get the upstream 304 from the `GET`, grab the body from the cache, and return it with a 200 to us.  Is that not what happens?

I currently suspect the cache library as the memory hog; it may need to be taught support for streamimg cache backends.

Reply:
I'm not exactly sure what is going on, but just issuing the GET, even if the cached file on disk, causes tons of memory usage that never gets GC'd. Adding this HEAD stops it. That's as far as I looked.